starwars/src/App.js

```diff
@@ -1,19 +1,58 @@
-import React from 'react';
-import './App.css';
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "./App.css";
+import ActorInfo from "./ActorsInfo";
+import styled from "styled-components";
 
+const starWarsApi = "https://swapi.co/api/people/";
 const App = () => {
   // Try to think through what state you'll need for this app before starting. Then build out
-  // the state properties here.
-
-  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
+  // the state properties here
+  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a
   // side effect in a component, you want to think about which state and/or props it should
   // sync up with, if any.
+  const [character, setCharacter] = useState([]);
+
+  const H1 = styled.h1`
+    color: #615961;
+    text-align: center;
+  `;
+  const Adiv = styled.div`
+    display: flex;
+    flex-wrap: wrap;
+    color: brown;
+  `;
+
+  useEffect(() => {
+    axios
+      .get(starWarsApi)
+      .then(response => {
+        setCharacter(response.data.results);
+      })
+      .catch(error => {
+        console.log("invalid Api");
```

Comment:
Don't forget about the sad path. Let users know if something goes bad when fetching data. "invalid Api" is your typed error message which give no clues even to you. What if the API is valid, but there's some internal error on the server? You would never know about this! Try to mess up API url and try this instead `console.log(error.message)`, see? How about displaying a separate component just for these cases if something goes wrong? 😃